### DISCUSSION TITLE
Continue Supporting Manifest v2

### DISCUSSION THREAD

**Discussion by Snuupy** [Alex313031/thorium#12] [Ideas]
Context: Chrome deprecated Manifest v2 to cripple adblockers. What patches (if any) would (now, or in the future) need to be submitted so Thorium can continue supporting Manifest v2 and uBlock Origin can continue to work?

**Comment by Alex313031** [comment]
@Snuupy I am aware of this and looking into it. Even if I can't figure out a patch to enable it forever, I can at least delay it by removing the deprecation flag that for right now is toggle-able. Why would uBlock Origin not work though? It is in active development, so the author would almost certainly migrate to V3. Unless there is something in the V3 spec that would inherently make ad blockers not work? So what do you mean they deprecated it to cripple adblockers?

**Comment by Alex313031** [comment]
@Snuupy Also, how do you like Thorium, and what platform(s) do you use it on?
I am so greatful with how my userbase has skyrocketed way beyond what I thought would happen. Mainly thanks to https://chromium.woolyss.com/

**Comment by Snuupy** [comment]
@Alex313031 

> the author would almost certainly migrate to V3

negative. https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-456179825 and https://github.com/uBlockOrigin/uBlock-issues/issues/338#issuecomment-456171375

> Unless there is something in the V3 spec that would inherently make ad blockers not work?

Exactly. afaik it's because it removes the ability to block network requests via extensions or limits it significantly.

> how do you like Thorium, and what platform(s) do you use it on?

I gave it a try because I was tired of user hostile browsers. Firefox didn't have the features that I wanted (namely SSBs - site specific browsers, nor a proper UI for profile switching). I'm on Windows myself. I had to switch back to Chrome thoguh because it didn't have profile sync or some other Google specific features last time I tried it but it's been at least a month or 2.

Edit: I see google sync is now there - I might just have to give it a go again :)

**Comment by Alex313031** [comment]
@Snuupy oooooo ok then, I will definitely look into this then!
Yeah the windows builds are in Thorium-Win.
I purchased a quota for 1 million hits per second on the google sync API Keys, so builds since M99 for all platforms now have google sync, translate, etc.
Latest Thorium win build works well, but there's a known bug that you will see in the releases: the icon in the taskbar when running is a yellow exclamation icon instead of the Thorium logo. It does not impact usablility. I'm working on a fix, and will be updating the logos anyway because Chrome and Chromium got new logos for the first time since 2014 with M100 (ChromiumOS only) and M101 (other platforms)

Thanks for alerting me to this manifest V3 information. I just thought it'll be another minor update like normal.

**Comment by Alex313031** [comment]
@Snuupy Man the more I'm reading about it this just sounds like an evil corporation type move. If I can't figure out a patch or someone else cant figure out a patch, and once the delay on the deprecation flag runs out, the last worst case scenario thing is that I will maintain a build of the last chromium version that will work with V2. I think I agree with you now that this might be an intentional thing to restrict ad blockers.

**Comment by Snuupy** [comment]
Been caught up with life so it took me a while to get back to this:

>purchased a quota for 1 million hits per second...all platforms now have google sync, translate, etc.

H Y P E

>Thanks for alerting me to this manifest V3 information. 

❤️ 

>sounds like an evil corporation type move

Yup. "Don't be evil" my ass.

>If I can't figure out a patch

There is definitely one floating around somewhere, I know Brave has implemented it. I don't have the mental bandwidth right now to dig for the link to the commit right now but I know it exists. 

>intentional thing to restrict ad blockers

Absolutely. Why else does chrome/all the chromium forks on android not have extension support? (Except kiwi/iceraven/firefox obviously)

**Comment by Alex313031** [comment]
@Snuupy If you could point me in some direction as far as the v3 patch, I would be happy to implement it in thorium.
Kiwi is chromium based, iceraven and firefox are firefox based. They are still limited in the extensions they can install though but.

**Reply by whatwareweb** [reply]
Just so you know, that extension limitation in firefox is artificial, to, guess what!! keep google from getting mad about firefox breaking the "no app stores inside of app stores" rule on Play store. Here's the article to get around that! [https://support.mozilla.org/en-US/kb/extended-add-support](https://support.mozilla.org/en-US/kb/extended-add-support)

**Comment by whatwareweb** [comment]
I'm not anywhere close to enough experience to touch such huge codebase as chromium but I do think this is an important topic going forward as Chrome is going to end support for this manifest V2 in June 2024. Brave has said they will support manifest V2, but I could not find any commits or PR's searching their repository.

**Comment by tz-shuhag** [comment]
@Alex313031  One option is that you can add adblock (ublock origin)  internally like Brave does even if google rolls out mv3. You can get something from here maybe - https://github.com/uazo/cromite

**Reply by KaKi87** [reply]
uBO is not the only extension affected by Manifest V3. Violentmonkey, Redirector, Hover Zoom+, all of those are extensions I will absolutely keep needing.

**Comment by denny466** [comment]
Here's how Brave force enables V2: https://github.com/brave/brave-core/blob/master/chromium_src/extensions/common/extension_features.cc

More info: https://brave.com/blog/brave-shields-manifest-v3/

Thank you for making Thorium! 🫶❤️

**Comment by gz83** [comment]
MV2 extensions are still supported.

**Reply by KaKi87** [reply]
Will that still be true for 151 though ?

**Reply by gz83** [reply]
I believe version 151 still supports it, but I would still recommend migrating to MV3 as soon as possible.

@KaKi87

**Reply by KaKi87** [reply]
Not everything can be migrated to MV3, and uBO is not the only one affected.

**Reply by ezxpro** [reply]
> MV2 extensions are still supported.

I don't think that's accurate. I'm on version 150.0.7871.114 of thorium and Manifest V3 UBlock Origin has disappeared from the come webstore (even though their github says it's still there for chromium users), and upon trying to install it manually I get this:
<img width="543" height="207" alt="image" src="https://github.com/user-attachments/assets/d98007b0-7a55-49ef-a0cd-c1b8f876fc0d" />

**Reply by gz83** [reply]
Thorium did not release this version; furthermore, MV3-based extensions will not be removed from or hidden on the CWS.

@ezxpro

**Reply by ezxpro** [reply]
@gz83 Ok, I don't know where I got that binary. Picked the latest release here and it worked fine, thanks.
